src/classes/STG_InstallScript.cls

```diff
@@ -338,24 +345,37 @@ global without sharing class STG_InstallScript implements InstallHandler {
         }
     }
     
-    /**********************************************************************************************************
-    * @description Enables prevent deletion by default on Hierarchy Custom Settings for new installs and 
-    * when upgrading to 1.95
+    /************************************************************************************************************************************************************************
+    * @description Enables Prevent Account Deletion and Plan Requirement Deletion by default on Hierarchy Custom Settings for new installs and when upgrading to 1.95
     * @return void
     */
     global static void enablePreventDeletionHierarchyCustomSettings(Version previousVersion) {
-        if ((previousVersion == NULL) || (previousVersion != null && previousVersion.compareTo(new Version(1,95)) < 0)) {
+        if ((previousVersion == null) || (previousVersion != null && previousVersion.compareTo(new Version(1,95)) < 0)) {
             UTIL_Debug.debug('***Upgrading to version 1.95 or new installs');
             Hierarchy_Settings__c orgSettings = UTIL_CustomSettingsFacade.getSettings();
             
             if (orgSettings.Prevent_Account_Deletion__c == null || orgSettings.Prevent_Account_Deletion__c != true) {
                 orgSettings.Prevent_Account_Deletion__c = true; 
             }
             if (orgSettings.Prevent_Plan_Requirement_Deletion__c == null || orgSettings.Prevent_Plan_Requirement_Deletion__c != true) {
-                orgSettings.Prevent_Plan_Requirement_Deletion__c = true; 
+                orgSettings.Prevent_Plan_Requirement_Deletion__c = true;
+            }
+            upsert orgSettings;
+        }
+    }
+
+    /*****************************************************************************************************************************************
+    * @description Enables Prevent Program Plan Deletion by default on Hierarchy Custom Settings for new installs and when upgrading to 1.96
+    * @return void
+    */
+    global static void enablePreventPPlanDeletionHierarchyCustomSettings(Version previousVersion) {
+        if ((previousVersion == null) || (previousVersion != null && previousVersion.compareTo(new Version(1,96)) < 0)) {
```

Comment:
@skosgi1 , I was wondering and maybe this is not a great way of thinking because we want to break things down modularly that 1 method does 1 thing, that instead of creating a 2nd enablePreventPPlanDeletionHierarchyCustomSettings method, that we add the if statement that checks if it is version 1.96 to enablePreventDeletionHierarchyCustomSettings method so that we only have 1? But again, I can see the argument that maybe we want 1 method to do 1 thing. @MatthewBlanski

Reply:
[enablePreventDeletionMethods.txt](https://github.com/SalesforceFoundation/EDA/files/4504932/enablePreventDeletionMethods.txt)

Reply:
After looking over the code and getting some context, I feel like since we have enablePreventDeletionHierarchyCustomSettings and it is global, we should leverage the method for how it is named. This aligns more closely with option 2 from what Kim provided. What I would prefer to see would be blocks for the 1.95 and 1.96 version that then contain individual methods for each discrete setting change and return the modified settings. Having the upsert be in the global method seems preferable in this case and should improve testability.